.travis.yml

```diff
@@ -15,7 +15,22 @@ addons:
     chrome: stable
 install:
     - npm install
-# install synapse prerequisites for end to end tests
-    - sudo apt-get install build-essential python2.7-dev libffi-dev python-pip python-setuptools sqlite3 libssl-dev python-virtualenv libjpeg-dev libxslt1-dev
-script:
-    ./scripts/travis.sh
+    - ./scripts/travis/install-deps.sh
+
+matrix:
+    include:
+        - name: Linting Checks
+          script:
+            # run the linter, but exclude any files known to have errors or warnings.
+            - npm run lintwithexclusions
```

Comment:
eslint dependency won't have been installed here yet?

Reply:
Note that build matrices inherit all the `install`/`script`/etc steps from the top level unless they're explicitly overridden, so in this case it'lll run the top level which doesn `./scripts/travis/install-deps.sh`